core/src/driver/stablex_driver.rs

```diff
@@ -189,17 +216,49 @@ impl<'a> StableXDriver for StableXDriverImpl<'a> {
     }
 }
 
+fn burnt_fees(orders: &[Order], solution: &Solution) -> Result<U256> {
+    let orders: HashMap<u16, &Order> = orders.iter().map(|order| (order.id, order)).collect();
+    let mut fee_token_imbalance = BigInt::zero();
+    for executed_order in solution.executed_orders.iter() {
+        let id = executed_order.order_id;
+        let order = orders
+            .get(&id)
+            .ok_or_else(|| anyhow!("order {} not found", id))?;
+        if order.sell_token == 0 {
+            fee_token_imbalance += executed_order.sell_amount;
+        }
+        if order.buy_token == 0 {
+            fee_token_imbalance -= executed_order.buy_amount;
+        }
+    }
+    bigint_u256::bigint_to_u256(&fee_token_imbalance)
+        .ok_or_else(|| anyhow!("invalid burnt fee amount: {}", fee_token_imbalance))
+}
+
+fn gas_price_cap(burnt_fees: U256, num_trades: usize) -> U256 {
+    // This is approximate. Depends on reversion of previous solution.
+    const GAS_PER_TRADE: f64 = 120_000.0;
+    const PRICE_OF_ETHER_IN_OWL: f64 = 200.0;
+    // The previous two values are approximations and we do not need to be economically viable at
+    // this point in time. So be more lenient with respect go gas prices.
+    const EXTRA_FACTOR: f64 = 10.0;
```

Comment:
nit: Let's call this `subsidy_factor` or the like (extra sounds very vague). It should probably also be passed in via a config so that we can adjust it without code changes.